verkle.go

```diff
@@ -0,0 +1,201 @@
+	var TxRoot [32]byte
+	var eth1Genesis *core.Genesis
+
+	// Load the Eth1 block from the Eth1 genesis config
+	if g.Eth1Config != "" {
+		eth1Genesis, err = loadEth1GenesisConf(g.Eth1Config)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Load the genesis timestamp from the CL config, this is better in terms of compatibility for shadowforks
+	if spec.MIN_GENESIS_TIME != 0 {
+		fmt.Println("Using CL MIN_GENESIS_TIME for genesis timestamp")
+
+		// Set beaconchain genesis timestamp based on config genesis timestamp
+		beaconGenesisTimestamp = spec.MIN_GENESIS_TIME
+	} else {
+		beaconGenesisTimestamp = g.Eth1BlockTimestamp
+	}
+
+	if g.ShadowForkEth1RPC != "" {
+		client, err := ethclient.Dial(g.ShadowForkEth1RPC)
+		if err != nil {
+			return fmt.Errorf("A fatal error occurred creating the ETH client %s", err)
+		}
+
+		// Get the latest block
+		blockNumberUint64, err := client.BlockNumber(ctx)
+		blockNumberBigint := new(big.Int).SetUint64(blockNumberUint64)
+		resultBlock, err := client.BlockByNumber(ctx, blockNumberBigint)
+		if err != nil {
+			return fmt.Errorf("A fatal error occurred getting the ETH block %s", err)
+		}
+
+		// Set the eth1Block value for use later
+		eth1Block = resultBlock
+
+		// Convert and set the difficulty as the prevRandao field
+		prevRandaoMix = bigIntToBytes32(eth1Block.Difficulty())
+
+		// Copy the txRoot from the block
+		copy(TxRoot[:], eth1Block.TxHash().Bytes())
+
+	} else if g.Eth1Config != "" {
+
+		// Generate genesis block from the loaded config
+		eth1Block = eth1Genesis.ToBlock()
+
+		// Set as default values
+		prevRandaoMix = common.Bytes32{}
+		TxRoot = common.PayloadTransactionsType(spec).DefaultNode().MerkleRoot(tree.GetHashFn())
+
+	} else {
+		fmt.Println("no eth1 config found, using eth1 block hash and timestamp, with empty ExecutionPayloadHeader (no PoW->PoS transition yet in execution layer)")
+		eth1BlockHash = g.Eth1BlockHash
+		execHeader = &verkle.ExecutionPayloadHeader{}
+	}
+
+	eth1BlockHash = common.Root(eth1Block.Hash())
+
+	extra := eth1Block.Extra()
+	if len(extra) > common.MAX_EXTRA_DATA_BYTES {
+		return fmt.Errorf("extra data is %d bytes, max is %d", len(extra), common.MAX_EXTRA_DATA_BYTES)
+	}
+
+	baseFee, _ := uint256.FromBig(eth1Block.BaseFee())
+
+	execHeader = &verkle.ExecutionPayloadHeader{
+		ParentHash:    common.Root(eth1Block.ParentHash()),
+		FeeRecipient:  common.Eth1Address(eth1Block.Coinbase()),
+		StateRoot:     common.Bytes32(eth1Block.Root()),
+		ReceiptsRoot:  common.Bytes32(eth1Block.ReceiptHash()),
+		LogsBloom:     common.LogsBloom(eth1Block.Bloom()),
+		PrevRandao:    prevRandaoMix,
+		BlockNumber:   view.Uint64View(eth1Block.NumberU64()),
+		GasLimit:      view.Uint64View(eth1Block.GasLimit()),
+		GasUsed:       view.Uint64View(eth1Block.GasUsed()),
+		Timestamp:     common.Timestamp(eth1Block.Time()),
+		ExtraData:     extra,
+		BaseFeePerGas: view.Uint256View(*baseFee),
+		BlockHash:     eth1BlockHash,
+
+		TransactionsRoot:     TxRoot,
```

Comment:
`withdrawalsRoot` is not set, even in the capella genesis its not set

Reply:
hmm, good catch! ill add it later today